xmtp_mls/src/storage/encrypted_store/group.rs

```diff
@@ -228,18 +235,14 @@ impl DbConnection {
         }
 
         if let Some(conversation_type) = conversation_type {
-            match conversation_type {
-                ConversationType::Group => {
-                    query = query.filter(groups_dsl::dm_inbox_id.is_null());
-                }
-                ConversationType::Dm => {
-                    query = query.filter(groups_dsl::dm_inbox_id.is_not_null());
-                }
-                ConversationType::Sync => {}
-            }
+            query = query.filter(groups_dsl::conversation_type.eq(conversation_type));
         }
 
-        query = query.filter(groups_dsl::purpose.eq(Purpose::Conversation));
+        // Were sync groups explicitly asked for? Was the include_sync_groups flag set to true?
+        // Otherwise filter sync groups out by default.
+        if !matches!(conversation_type, Some(ConversationType::Sync)) && !include_sync_groups {
+            query = query.filter(groups_dsl::conversation_type.ne(ConversationType::Sync));
+        }
```

Comment:
Aside from removing Purpose and adding ConversationType to the db, here's the main change of this PR.
No longer unconditionally filtering out Sync groups, however we still filter them out by default.